gio-subclass/tests/simple_application.rs

```diff
@@ -0,0 +1,189 @@
+
+use gio_subclass::application::*;
+
+
+
+mod imp {
+    use super::*;
+
+    pub struct SimpleApplication {
+    }
+
+    static PROPERTIES: [Property; 0] = [];
+
+    impl SimpleApplication {
+        pub fn get_type() -> glib::Type {
+            static ONCE: Once = ONCE_INIT;
+            static mut TYPE: glib::Type = glib::Type::Invalid;
+
+            ONCE.call_once(|| {
+                let t = register_type(SimpleApplicationStatic);
+                unsafe {
+                    TYPE = t;
+                }
+            });
+
+            unsafe { TYPE }
+        }
+
+        fn class_init(klass: &mut ApplicationClass) {
+            klass.install_properties(&PROPERTIES);
+        }
+
+        fn init(obj: &Application) -> Box<ApplicationImpl<Application>> {
+            let imp = Self {
+            };
+            Box::new(imp)
+        }
+    }
+
+    impl ObjectImpl<Application> for SimpleApplication {
+        fn set_property(&self, obj: &glib::Object, id: u32, value: &glib::Value) {
+            let prop = &PROPERTIES[id as usize];
+
+            match *prop {
+                _ => unimplemented!(),
+            }
+        }
+
+        fn get_property(&self, _obj: &glib::Object, id: u32) -> Result<glib::Value, ()> {
+            let prop = &PROPERTIES[id as usize];
+
+            match *prop {
+                _ => unimplemented!(),
+            }
+        }
+    }
+
+    impl ApplicationImpl<Application> for SimpleApplication{
+
+
+        fn local_command_line(&self, application: &Application, arguments: &mut ArgumentList) -> Option<i32> {
+
+            let mut rm = Vec::new();
+
+            for (i, line) in arguments.iter().enumerate() {
+                if line.starts_with("--local-"){
+                    println!("handling argument {:?} locally", line);
+                    rm.push(i)
+                }else{
+                    println!("not handling argument {:?} locally", line);
+
+                }
+            }
+
+            rm.reverse();
+
+            for i in rm.iter(){
+                arguments.remove(*i);
+            }
+
+            None
+        }
+
+        fn command_line(&self, application: &Application, cmd_line: &gio::ApplicationCommandLine) -> i32 {
+
+            let arguments = cmd_line.get_arguments();
+
+            for arg in arguments{
+                println!("handling argument {:?} remotely", arg);
+            }
+
+            return 0;
+        }
+
+    }
+
+    struct SimpleApplicationStatic;
+
+    impl ImplTypeStatic<Application> for SimpleApplicationStatic {
+        fn get_name(&self) -> &str {
+            "SimpleApplication"
+        }
+
+        fn new(&self, obj: &Application) -> Box<ApplicationImpl<Application>> {
+            SimpleApplication::init(obj)
+        }
+
+        fn class_init(&self, klass: &mut ApplicationClass) {
+            SimpleApplication::class_init(klass);
+        }
+    }
+}
+
+glib_wrapper! {
+    //TODO: is this correct???
```

Comment:
You also want to add the two interfaces here, like in the `glib_wrapper!` usage above